lib/compiler/passes/generate-js.js

```diff
@@ -1373,6 +1372,14 @@ function generateJS(ast, options) {
       "        : peg$computeLocation(peg$maxFailPos, peg$maxFailPos)",
       "    );",
       "  }",
+      "  if (options.soft) {",
+      "     return {result: peg$result, success, fail};",
```

Comment:
If `success` is true, maybe `fail` should be undefined?  Calling it will always give unexpected results on success.